### PR TITLE
Use separate thread pool for IO operations for backups

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Introduced a separate thread pool for backup IO operations. This will allow to scale it independently from other pool and increase performance.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: src/Core/ServerSettings.h [hunk 1]
```diff
@@ -21,6 +21,9 @@ namespace DB
     M(UInt64, max_io_thread_pool_size, 100, "The maximum number of threads that would be used for IO operations", 0) \
     M(UInt64, max_io_thread_pool_free_size, 0, "Max free size for IO thread pool.", 0) \
     M(UInt64, io_thread_pool_queue_size, 10000, "Queue size for IO thread pool.", 0) \
+    M(UInt64, max_backuos_io_thread_pool_size, 1000, "The maximum number of threads that would be used for IO operations for BACKUP queries", 0) \
+    M(UInt64, max_backups_io_thread_pool_free_size, 0, "Max free size for backups IO thread pool.", 0) \
+    M(UInt64, backups_io_thread_pool_queue_size, 10000, "Queue size for backups IO thread pool.", 0) \
```
Comment (vitlibar) [inline src/Core/ServerSettings.h:26]: Can't we just set that to `0` (which means "unlimited size of the queue")

File: src/Core/ServerSettings.h [hunk 2]
```diff
@@ -21,6 +21,9 @@ namespace DB
     M(UInt64, max_io_thread_pool_size, 100, "The maximum number of threads that would be used for IO operations", 0) \
     M(UInt64, max_io_thread_pool_free_size, 0, "Max free size for IO thread pool.", 0) \
     M(UInt64, io_thread_pool_queue_size, 10000, "Queue size for IO thread pool.", 0) \
+    M(UInt64, max_backuos_io_thread_pool_size, 1000, "The maximum number of threads that would be used for IO operations for BACKUP queries", 0) \
```
Comment (antonio2368) [inline src/Core/ServerSettings.h:24]:
```suggestion
    M(UInt64, max_backups_io_thread_pool_size, 1000, "The maximum number of threads that would be used for IO operations for BACKUP queries", 0) \
```

Comment (nikitamikhaylov) [reply src/Core/ServerSettings.h]: Thanks

File: src/Core/ServerSettings.h (hunk 1 above)
Comment (nikitamikhaylov) [reply src/Core/ServerSettings.h]: Yes.

**Pull request merged**